internal/appsec/dyngo/instrumentation/httpinstr/http.go

```diff
@@ -117,6 +84,46 @@ func WrapHandler(handler http.Handler, span ddtrace.Span) http.Handler {
 	})
 }
 
+// MakeHandlerOperationArgs  creates the HandlerOperationArgs out of a standard
+// http.Request, including the given span.
+func MakeHandlerOperationArgs(r *http.Request, span ddtrace.Span) HandlerOperationArgs {
+	headers := make(http.Header, len(r.Header))
+	for k, v := range r.Header {
+		k := strings.ToLower(k)
+		if k == "cookie" {
+			// Do not include cookies in the request headers
+			continue
+		}
+		headers[k] = v
+	}
+	var cookies map[string][]string
+	if reqCookies := r.Cookies(); len(reqCookies) > 0 {
+		cookies = make(map[string][]string, len(reqCookies))
+		for _, cookie := range reqCookies {
+			if cookie == nil {
+				continue
+			}
+			cookies[cookie.Name] = append(cookies[cookie.Name], cookie.Value)
+		}
+	}
+	host := r.Host
+	headers["host"] = []string{host}
+	return HandlerOperationArgs{
+		Span:       span,
+		IsTLS:      r.TLS != nil,
+		Method:     r.Method,
+		Host:       r.Host,
+		Path:       r.URL.Path,
+		RequestURI: r.RequestURI,
+		RemoteAddr: r.RemoteAddr,
+		Headers:    headers,
+		Cookies:    cookies,
```

Comment:
Why not take the `r.Headers` field as it is here and use the same type (`http.Header`) ? Same for the cookies. It would be more standardised and you'd avoid creating new maps and loops.